core/src/main/java/com/yahoo/imapnio/async/request/AbstractFolderActionCommand.java

```diff
@@ -41,6 +41,24 @@ public void cleanup() {
         this.folderName = null;
     }
 
+    /**
+     * Get the op name.
+     *
+     * @return operation name
+     */
+    public String getOp() {
```

Comment:
can you create an abstract OpenFolderCommand so these 2 classes extend from that.  Lets not have ExamineFolderCommand and SelectFolderCommand extend from AbstractFolderActionCommand any more (so u do not need to expose getOp and getFolderName at all).  AbstractFolderActionCommand is provided for simple folder action commands.